youtube_dl/YoutubeDL.py

```diff
@@ -965,6 +966,9 @@ def __process_playlist(self, ie_result, download):
 
         self.to_screen('[download] Downloading playlist: %s' % playlist)
 
+        if self.params.get('autonumber_reset'):
+            self._num_downloads = 0
```

Comment:
This will break `--max-download`.

Reply:
added a new property `_autonumber_index` to track separately from `_num_downloads` to prevent `--max-download` from breaking